ow_plexil/src/plans/FaultDependenciesDemo.plp

```diff
@@ -0,0 +1,108 @@
+// The Notices and Disclaimers for Ocean Worlds Autonomy Testbed for Exploration
+// Research and Simulation can be found in README.md in the root directory of
+// this repository.
+
+// The Antenna will Pan, Tilt and then take a picture. If the Pan Fault is active
+// it will only Tilt and Take a Picture. If the Tilt Fault is active it will only 
+// Pan and take a picture. If both Pan and Tilt are inoperable, the backup action
+// will begin and unstow/stow the arm repeatably. Camera faults will prevent a picture
+// from being taken until they are cleared. 
+
+// NOTE: As of 10/18/2023 The backup action should be robust to faults, but due to the
+// SynchronousCommand bug it will freeze the whole plexil plan if an arm fault is 
+// activated during execution. This is being investigated.
+
+
+
+ #include "plan-interface.h"
+
+FaultDependenciesDemo:
+{
+
+  Real NewPanAngle = 0;
+  Real NewTiltAngle = 20;
+  Real Count = 0;
+  Boolean BackupAction = false;
+  Boolean CameraFault = false;
+  RepeatCondition true;
+
+  log_info ("Starting FaultDependenciesDemo plan...");
+
+  Run: Concurrence
+  {
+    RepeatCondition true;
+    Wait 1;
+
+    PanTiltPattern: Sequence
+    {
+      StartCondition Lookup(IsOperable("Antenna")) && !BackupAction;
+      RepeatCondition true;
+      Wait 2;
+
+      NewPanAngle = (Lookup(PanDegrees) + 15) mod 360;
+      if NewPanAngle > 180 {
+        NewPanAngle = NewPanAngle - 360;
+      }
+      if Count % 2 == 0{
+        NewTiltAngle = 10;
+      }
+      else{
+        NewTiltAngle = -10;
+      }
+      Count = Count + 1;
+
+      if Lookup(IsOperable("Pan")) && Lookup(IsOperable("Tilt")) {
+        log_info("Panning and Tilting...");
+        LibraryCall PanTiltMoveJoints (PanDegrees=NewPanAngle, TiltDegrees=NewTiltAngle);
+      }
+      elseif Lookup(IsOperable("Tilt")){
+        log_info("Pan Inoperable, executing Tilt only...");
+        LibraryCall Tilt (Degrees=NewTiltAngle);
+      }
+      elseif Lookup(IsOperable("Pan")){
+        log_info("Tilt Inoperable, executing Pan only...");
+        LibraryCall Pan (Degrees=NewPanAngle);
+      }
+      else{
+        log_info("Tilt and Pan Inoperable, beginning backup pattern...");
+        BackupAction = true;
+      }
+
+      if Lookup(IsOperable("CameraCapture")){
+        log_info("Taking Picture...");
+        LibraryCall CameraCapture();
+      }
+
+      if CameraFault{
+        LibraryCall CameraCapture();
+        if Lookup(IsOperable("CameraCapture")){
+          log_info("Camera now Operable, picture taken");
+          CameraFault = false;
+        }
+        else{
+          CameraFault = true;
+          log_info("WARNING: Camera Is Inoperable, no picture taken.");
+        }
+      }
+    }
```

Comment:
Yes, if we dont have a fault it can be used normally. Otherwise we need to test if the fault is still active by calling CameraCapture and checking if it is now operable.